source/connect.txt

```diff
@@ -9,5 +9,147 @@ Connect to a Deployment
 .. contents:: On this page
    :local:
    :backlinks: none
-   :depth: 1
+   :depth: 2
    :class: singlecol
+
+
+Ensure that a MongoDB deployment is running before attempting to 
+connect to it with ``mongosh``. The connection method you use depends 
+on your deployment type.
+
+Local MongoDB Instance on Default Port
+--------------------------------------
+
+You can run ``mongosh`` without any command-line options
+to connect to a MongoDB instance running on your **localhost** with 
+**default port** 27017:
+
+.. code-block:: sh
+
+   mongosh
+
+Local MongoDB Instance on a Non-Default Port
+--------------------------------------------
+
+To specify a port to connect to on localhost, use the 
+:manual:`connection string </reference/connection-string>` format. For 
+example, to connect to a MongoDB instance running on localhost with a 
+non-default port 28015:
+
+.. code-block:: sh
+
+   mongosh "localhost:28015"
+
+MongoDB Instance on a Remote Host
+---------------------------------
+
+To specify a remote host and port, use a :manual:`connection string
+</reference/connection-string>`. For example, to connect to a MongoDB
+instance running on a remote host machine on port 28015:
+
+  .. code-block:: sh
+
+     mongosh "mongodb://mongodb0.example.com:28015"
+
+  .. note:: Connecting to |service|
+
+     If your remote host is a |service-fullname| cluster, you can copy 
+     your connection string from the |service| UI. To learn more, see 
+     :atlas:`Connect to a Cluster <connect-to-cluster/#use-the-connect-dialog-to-connect-to-your-cluster>`.
+
+Connection Considerations
+-------------------------
+
+Authentication
+~~~~~~~~~~~~~~
+
+To connect to a MongoDB instance requires authentication:
+
+- You can specify the username, password, and authentication database 
+  in the :manual:`connection string </reference/connection-string>`. 
+  For example, to connect to a remote MongoDB instance and
+  authenticate against the ``admin`` database as user ``alice``:
+
+  .. code-block:: sh
+
+     mongosh "mongodb://alice:<password>@mongodb0.examples.com:28015/?authSource=admin"
+
+- You can use the ``--username``, ``--password``, and
+  ``--authenticationDatabase`` command-line options. For example, to
+  connect to a remote MongoDB instance and authenticate against the 
+  ``admin`` database as user ``alice``:
+
+  .. code-block:: sh
```

Comment:
Can we adjust this example to discourage users from typing their passwords in plaintext? Let's instead direct them to just specify --username and say they will be prompted for a password.

Maybe then use the note below to call out the --password option, if they want to use that.

Reply:
Will do, that makes sense - I'm reading this as remove the connection string option entirely (since you can't connect without providing the password there, too). I'll save that copy in case that's not what you had in mind.